### PR TITLE
[ENH] add golden-output regression test for `TimesFM2Forecaster`

### PR DESCRIPTION
**Pull request by Nischal1425**
## Reference Issues/PRs
Part of #10672

## What does this implement/fix?
Adds a regression test that validates `TimesFM2Forecaster` predictions against the raw `transformers.TimesFm2_5ModelForPrediction` implementation, called directly and independently of sktime, following the pattern established in #10627 for `TinyTimeMixerForecaster`.

Reference values were generated with this script (also embedded as a
comment in the test file for reproducibility):

    from transformers import TimesFm2_5ModelForPrediction
    import numpy as np, torch
    from sktime.datasets import load_airline

    y_train = load_airline().iloc[:-12]
    model = TimesFm2_5ModelForPrediction.from_pretrained(
        "google/timesfm-2.5-200m-transformers", device_map="cpu", dtype=None
    )
    model.eval()
    past_values = torch.from_numpy(
        np.expand_dims(y_train.values.astype(np.float32), axis=0)
    ).to(model.dtype).to(model.device)
    with torch.no_grad():
        output = model(past_values=past_values)
    preds = output.mean_predictions.ravel().detach().float().cpu().numpy()
    preds[:3]  # -> [407.3865051269531, 378.2121887207031, 429.0090026855469]

`sktime`'s wrapper (`TimesFM2Forecaster._predict`) is a thin pass-through to this same model call (reshape context to `(1, seq_len)`, cast dtype/device, forward pass, take `mean_predictions`), so this test validates that the wrapper introduces no discrepancy versus the source implementation - the same class of regression #10626/#10627 caught for TTM.

Verified: sktime's output matches the raw-transformers reference to full float32 precision (exact match, not just within tolerance). Also confirmed the test actually discriminates, reran it with a deliberately wrong reference value and it failed as expected.

## Does your contribution introduce a new dependency?
No.

## Did you add any tests for the change?
Yes — this PR is itself a new regression test, `test_timesfm2_airline_predictions_match_transformers_reference`, gated behind `_check_soft_dependencies` so it's skipped when `transformers[torch]` isn't installed.

### CODE REVIEW

File: sktime/forecasting/tests/test_timesfm2.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    407.3865051269531,
+    378.2121887207031,
+    429.0090026855469,
+]
+
+
+@pytest.mark.skipif(
+    not _check_soft_dependencies("transformers[torch]>=4.52.0", severity="none"),
+    reason="run test only if TimesFM2Forecaster soft dependencies are present",
+)
```
Comment (geetu040) [inline sktime/forecasting/tests/test_timesfm2.py:51]: better to use `_check_estimator_deps` here

**Review by geetu040 — CHANGES_REQUESTED** [review]
Nice, we just need a few changes:
- also include the checkpoint [google/timesfm-2.0-500m-pytorch](https://huggingface.co/google/timesfm-2.0-500m-pytorch)
- update the tag `tests:specific` for `TimesFM2Forecaster` to include this test file

**Comment by Nischal1425** [comment]
Added the `google/timesfm-2.0-500m-pytorch` checkpoint as a second parametrized case, following the same pattern `test_ttm.py` uses. Reference values generated the same way as the 2.5 checkpoint (raw `transformers.TimesFmModelForPrediction` call, bypassing sktime) - sktime's wrapper matches exactly for both checkpoints.

**Review by geetu040 — CHANGES_REQUESTED** [review]
https://github.com/sktime/sktime/pull/10674#pullrequestreview-4797151469 not completely addressed

**Comment by Nischal1425** [comment]
Missed the `tests:specific` part on the first pass, sorry - added now: `"tests:specific": ["sktime.forecasting.tests.test_timesfm2"]`, matching the pattern `TinyTimeMixerForecaster` already uses. Verified via `_get_estimator_specific_test_modules("TimesFM2Forecaster")` that it resolves correctly.

**Review by geetu040 — DISMISSED** [review]
Thanks, LGTM!

CC: @fkiraly for merging this.

**Review by fkiraly — CHANGES_REQUESTED** [review]
The run conditions for the added tests are incorrect, FYI @geetu040.

This should be `run_tests_for_class`, not `_check_soft_dependencies`.

**Comment by geetu040** [comment]
> The run conditions for the added tests are incorrect, FYI @geetu040.
> 
> This should be `run_tests_for_class`, not `_check_soft_dependencies`.

@fkiraly  I see, I did leave this as a review comment above in https://github.com/sktime/sktime/pull/10674#discussion_r3665432802 to use `_check_estimator_deps` not `_check_soft_dependencies`, but I thought it was non-blocking, but agreed `run_tests_for_class` makes more sense and should be preferred.

**Comment by Nischal1425** [comment]
Switched to `run_test_for_class(TimesFM2Forecaster)` in 3e6e868, matching the pattern `test_arch.py` uses. This also fixes it properly rather than just addressing the letter of the request, the old `_check_soft_dependencies` check only gated on deps being installed, `run_test_for_class` additionally respects the `tests:vm=True` tag and incremental test-selection, which is the behavior the rest of the estimator's tests already get.
@fkiraly @geetu040 ready for another look.

**Review by fkiraly — APPROVED** [review]
Thanks!
